### PR TITLE
Mailbox/set should handle isSubscribed

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 1]
```diff
@@ -267,6 +267,128 @@ trait MailboxSetMethodContract {
...
+        .statusCode(SC_OK)
+        .contentType(JSON)
+        .extract
+        .body
+        .asString
+
+    Assertions.assertThat(server.getProbe(classOf[MailboxProbeImpl])
+      .listSubscriptions(BOB.asString())).contains("myMailbox")
+  }
+
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:391]: Do we need or not to add a test to check that the response does not contain the `isSubscribed` property? As the specs say we must omit properties in the response that were set in the request

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:391]: This seems a bit overkill to me to be honnest.

This is harder as we need to dynamically filter the response depending on what is specified by the client.

Maybe we should open another ticket for that?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[7efccfb3c71818b67efdba240362e67e5b316caa] Error response from daemon: No such container: james-parent-test-mailbox-set-subscribed-404fc91e-1f40-4a5d-a4ad-64a67fd86abf
```

Can be considered :green_apple: 

test this please

**Comment by chibenwa** [comment]
```
RabbitMQDeletedMessageVaultIntegrationTest>DeletedMessageVaultIntegrationTest.vaultEndpointShouldRestoreImapDeletedMailbox:252 » ConditionTimeout
```

test this please

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**